### PR TITLE
cwrsync: Add 6.2.3

### PR DESCRIPTION
**Pull request by CaselIT**
Readds rsync that seem to be again free. It was removed by https://github.com/ScoopInstaller/Scoop/commit/dac9d8313dfafda25b58d4de8cf2409798bb7859 since it was no longer available

Closes https://github.com/ScoopInstaller/Main/issues/2897

### CODE REVIEW

File: bucket/rsync.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+    "homepage": "https://itefix.net/cwrsync",
+    "version": "6.2.3",
+    "description": "rsync is a file transfer program capable of efficient remote update via a fast differencing algorithm.",
+    "license": "Itefix free software license (Simplified BSD)",
+    "##": "See also: https://www.itefix.net/content/cwrsync-licenseversion",
```
Comment (rashil2000) [inline bucket/rsync.json:6]:
```suggestion
    "license": "https://itefix.net/simplified-bsd-license",
```

**Review by rashil2000 — CHANGES_REQUESTED** [review]

File: bucket/rsync.json (hunk 1 above)
Comment (pratikpc) [reply bucket/rsync.json:6]: I believe we should also provide the URL of the license

Comment (rashil2000) [reply bucket/rsync.json:6]: This is the license URL - https://itefix.net/simplified-bsd-license

Comment (CaselIT) [reply bucket/rsync.json]: thanks. that part was copied from the old file at https://github.com/ScoopInstaller/Scoop/commit/dac9d8313dfafda25b58d4de8cf2409798bb7859

I've updated it

**Comment by pratikpc** [comment]
Personally I am of the opinion that we should try to support the MSYS based rsync instead. And a PR would probably be ready when Scoop can support ZSTD.

This rsync flip-flops between paid and free.

**Comment by rashil2000** [comment]
MSYS based rsync is not readily available. Personally I think isolating binaries that depend on msys-2.0.dll isn't a very compatible idea.

Besides, I think cwrsync has been free for more than a year now (from when I last used it). The MSYS version shouldn't be a blocker for this anyway.

**Review by rashil2000 — APPROVED** [review]

**Comment by pratikpc** [comment]
> The MSYS version shouldn't be a blocker for this anyway.

The package name is what IMHO is a blocker.

If this was cwrsync.json on the other hand, I would have no issues because an MSYS based rsync could obtain the rsync.json name.

Their Wikipedia article is also titled [cwRsync](https://en.wikipedia.org/wiki/CwRsync)

**Comment by CaselIT** [comment]
I can rename if you prefer. It's fine either way for me

File: bucket/rsync.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://itefix.net/dl/free-software/cwrsync_$version_x64_free.zip"
+            }
+        },
+        "hash": {
+            "url": "$url.sha256.asc"
+        },
+        "extract_dir": "cwrsync_$version_x64_free"
```
Comment (pratikpc) [inline bucket/rsync.json:27]: extract_dir must be within 64bit block

File: bucket/rsync.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "version": "6.2.3",
+    "description": "rsync is a file transfer program capable of efficient remote update via a fast differencing algorithm.",
+    "license": "https://itefix.net/simplified-bsd-license",
+    "architecture": {
+        "64bit": {
+            "url": "https://itefix.net/dl/free-software/cwrsync_6.2.3_x64_free.zip",
+            "hash": "19356b4910e587a158b3cffb1aa4cb6a942826b5727d97522ff9ee5f97236158"
+        }
+    },
+    "extract_dir": "cwrsync_6.2.3_x64_free",
```
Comment (pratikpc) [inline bucket/rsync.json:12]: extract_dir must be within 64bit block

**Review by pratikpc — CHANGES_REQUESTED** [review]

**Comment by rashil2000** [comment]
Okay, let's rename this to cwrsync.json

File: bucket/rsync.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+    "homepage": "https://itefix.net/cwrsync",
+    "version": "6.2.3",
```
Comment (pratikpc) [inline bucket/rsync.json:3]: version is usually at the top

Check https://github.com/ScoopInstaller/Main/blob/master/bucket/abc.json for the order

**Review by pratikpc — CHANGES_REQUESTED** [review]
extract_dir and incorrect order

File: bucket/rsync.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        "url": "https://itefix.net/cwrsync",
+        "regex": "cwrsync_([\\d.]+)_x64_free\\.zip"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://itefix.net/dl/free-software/cwrsync_$version_x64_free.zip"
+            }
+        },
+        "hash": {
```
Comment (pratikpc) [inline bucket/rsync.json:24]: hash should also be within the 64bit block

**Review by pratikpc — CHANGES_REQUESTED** [review]

**Comment by pratikpc** [comment]
@CaselIT hey please run checkver.ps1 offline once

https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate#using-checkverps1-to-query-and-autoupdate

**Comment by CaselIT** [comment]
Thanks for the review, I'll update

> @CaselIT hey please run checkver.ps1 offline once
> 
> https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate#using-checkverps1-to-query-and-autoupdate

I did and it was happy also with the current attribute position. (I copied them from another manifest)

In any case I'll update since it seem to make more sense

**Comment by CaselIT** [comment]
done. checkver is also happy:
```
cwrsync: 6.2.3 (scoop version is 6.2.2) autoupdate available
Autoupdating cwrsync
Searching hash for cwrsync_6.2.3_x64_free.zip in https://itefix.net/dl/free-software/cwrsync_6.2.3_x64_free.zip.sha256.asc
Found: 19356b4910e587a158b3cffb1aa4cb6a942826b5727d97522ff9ee5f97236158 using Extract Mode
Writing updated cwrsync manifest
```

File: .gitignore [hunk 6]
```diff
@@ -3,3 +3,4 @@
 /checkver*
 .idea
 *.iml
+.vscode
```
Comment (pratikpc) [inline .gitignore:6]: Please don't modify .gitignore

File: bucket/cwrsync.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+            "64bit": {
+                "url": "https://itefix.net/dl/free-software/cwrsync_$version_x64_free.zip",
+                "extract_dir": "cwrsync_$version_x64_free",
+                "hash": {
+                    "url": "$url.sha256.asc"
+                }
+            }
+        }
+    }
+}
```
Comment (pratikpc) [inline bucket/cwrsync.json:29]: Newline missing at the end of the file

**Review by pratikpc — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 6 above)
Comment (CaselIT) [reply .gitignore:6]: .idea is here, I think it makes sense to also have an ignore for .vscode, since it a very popular editor

Comment (pratikpc) [reply .gitignore:6]: For something similar to that, you would have to create a new PR.

**Review by pratikpc — CHANGES_REQUESTED** [review]
Please do not modify .gitignore in this PR

**Comment by CaselIT** [comment]
> 
> 
> Please do not modify .gitignore in this PR

whatever..

File: .gitignore (hunk 6 above)
Comment (CaselIT) [reply .gitignore]: created https://github.com/ScoopInstaller/Main/pull/2898

**Review by rashil2000 — APPROVED** [review]

**Review by pratikpc — APPROVED** [review]
Manifest looks okay

@rashil2000 uses cwrsync so might know better

**Comment by CaselIT** [comment]
for what is worth I've also tried installing it locally by adding the file to the scoop checkout and it did install correctly

**Comment by CaselIT** [comment]
thanks!

**Pull request merged**